### DISCUSSION TITLE
Compiler is out of heap space - MSVC 2017

### DISCUSSION THREAD

**Discussion by ssooffiiaannee** [pocoproject/poco#4820] [General]
OS : Windows 11

```
cd poco
cd cmake-build
cmake .. -G "Visual Studio 15 2017"
cmake --build . --config Release
```
I get the following build error : 
```
  ODBCStatementImpl.cpp
  Parameter.cpp
  Preparator.cpp
  SessionImpl.cpp
  TypeInfo.cpp
E:\poco\Foundation\include\Poco/Tuple.h(3128): fatal error C1060: compiler is out of heap space [E:\poco\cmake-build\Data\ODBC\DataODBC.vcxproj]
  Unicode.cpp
```
I tried lowering the memory allocation limit with the /Zm flag, as more heap space memory is allocated, but I still can't build the project.
This [link](https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/fatal-error-c1060?view=msvc-170) suggest the following : 
> Eliminate unnecessary include files.
> 
> Eliminate unnecessary global variables, for example, by allocating memory dynamically instead of declaring a large array.
> 
> Eliminate unused declarations.
> 
> Split the current file into smaller files.

**Comment by matejk** [comment]
Perhaps this is the problem:

https://developercommunity.visualstudio.com/t/c1060-with-nested-structures/329102

Or this:

https://developercommunity.visualstudio.com/t/158-regression-fatal-error-c1060-compiler-is-out-o/296784

Did you try switching to a newer Visual Studio?

**Comment by ssooffiiaannee** [comment]
Among the prerequisites is Visual C++ 17, using another version solves the "out of heap space" issue, but a simple example fails to link to a inexistant library

> LINK : fatal error LNK1104: cannot open file 'PocoFoundationd.lib' [D:\testing_poco\cmake-build\mongo_ex.vcxproj]

I have had a similar issue [check conversation](https://github.com/pocoproject/poco/discussions/4810), so I avoid using different VS versions

**Comment by ssooffiiaannee** [comment]
So the issue with missing PocoFoundationd.lib is maybe due to the fact that I built Poco in Release mode (default), and tried to build my example in Debug mode, or the opposite.
